README.md

```diff
@@ -600,25 +700,256 @@ uninstall_package
+
+![spack module av](.images/module_av.png)
+
+Those are modules for all software that we have installed so far! If you look at module names carefully then you can see :
+
+* modules are named as `package name` + `version` + `compiler` + `hash`
+* even external packages (installed by brew) have modules
+* lot of redundant modules for end users : why they need autoconf, automake module?
+* hash in module name is not useful for end-users (module names will change once version / dependency changes)
+
+These auto-generated modules are difficult to use for end users and that's where Spack support for modules comes handy! Details information is [here](http://spack.readthedocs.io/en/latest/module_file_support.html#).
+Below instructions provide quick instructions for generating module on OS X.
+
+We will look into `Environment Modules` as well as `LMod`. As discussed in `OS X` section, make sure you have `module` and `lmod` installed via brew. Your `.bashrc` or `.bash_profile` should activate the support for modules as:
+
+```
+MODULES_HOME=`brew --prefix modules`
+source ${MODULES_HOME}/Modules/init/bash
+```
+
+> brew install `lmod`/ `modules` under `/usr/local/opt/`
+
+In order to tune the module generation first step is to create `modules.yaml` in `$HOME/.spack/modules.yaml`. This configuration file controls how modules will be generated. Here is sample configuration for tcl modules:
+
+```
+modules:
+
+  enable::
+      - tcl
+  tcl:
+    all:
+      suffixes:
+          '^openmpi': 'openmpi'
+          '^python': 'python'
+
+      autoload: 'direct'
+      conflict:
+          - '${PACKAGE}'
+      environment:
+        set:
+          '${PACKAGE}_ROOT': '${PREFIX}'
+      filter:
+        environment_blacklist: ['CPATH', 'LIBRARY_PATH']
+
+    hash_length: 0
+    naming_scheme: '${PACKAGE}/${VERSION}-${COMPILERNAME}'
+
+    blacklist: ['cmake', 'hdf5', 'zlib', 'autoconf', 'libtool', 'pkg-config', 'automake', '%gcc@4.4', 'mod2c', '-compile']
+
+    openmpi:
+      environment:
+        set:
+          SLURM_MPI_TYPE: pmi2
+          OMPI_MCA_BTL_OPENIB_WARN_DEFAULT_GID_PREFIX: '0'
+```
+
+Here is brief explanation of the above modules configuration:
+
+* we are enabling tcl modules generation
+* if some package is dependent on openmpi then the module name of that package will have suffix `openmpi`. Same for python.
+* we want load all dependent modules automatically (e.g. if hdf5 is depend on zlib : if someone load hdf5 then load zlib module as well
```

Comment:
> if hdf5 is depend on zlib : if someone load hdf5 then load zlib module as well

I am not sure to understand the sentence. Do you mean: as *hdf5 depends on zlib, if hdf5 module is loaded then load zlib as well* ?

Reply:
yes, updated with more clarity.

(if the packages are CMake based then spack rpath dependencies but not sure about all packages. typical packaging stuff in scientific computing community)